subsys/bluetooth/host/crypto.c

```diff
@@ -114,6 +115,44 @@ int bt_rand(void *buf, size_t len)
 
 	return -EIO;
 }
+#else /* !CONFIG_BT_HOST_CRYPTO_RANDOM */
+int bt_rand(void *buf, size_t len)
+{
+	int ret, size;
+	size_t i = 0;
+
+	/* Check first that HCI_LE_Rand is supported */
+	if (!BT_CMD_TEST(bt_dev.supported_commands, 27, 7)) {
+		return -ENOTSUP;
+	}
+
+	while (len) {
+		struct bt_hci_rp_le_rand *rp;
+		struct net_buf *rsp;
+
+		ret = bt_hci_cmd_send_sync(BT_HCI_OP_LE_RAND, NULL, &rsp);
+		if (ret) {
+			return ret;
+		}
+
+		size = MIN(len, sizeof(rp->rand));
+
+		rp = (void *)rsp->data;
+		if (rp->status) {
+			return -EIO;
+		}
+
+		(void)memcpy((uint8_t *)buf + i, rp->rand, size);
```

Comment:
```suggestion
		rp = (void *)rsp->data;
		if (rp->status) {
			return -EIO;
		}

		size = MIN(len, sizeof(rp->rand));
		(void)memcpy((uint8_t *)buf + i, rp->rand, size);
```

Minor comment: We don't need to calculate size if rp->status is an error

Reply:
Thank you for your careful consideration.